tests/unit/client.py

```diff
@@ -262,28 +268,69 @@ def avoid_network_race_condition():
     for timeout in range(10):
         que = db.session.query(NetworkingAction).count()
         if (que > 0):
-            time.sleep(1)
+            time.sleep(.5)
```

Comment:
I think this is too short a time interval. I did some tests with different time intervals. It usually takes like 6-ish seconds. 
I think 1 sec is a good interval. Just saying,  no need to change code more than necessary.